src/iface/interface/mod.rs

```diff
@@ -82,6 +82,16 @@ pub struct Interface {
     fragmenter: Fragmenter,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum MulticastGroupState {
+    /// Joining group, we have to send the join packet.
+    Joining,
+    /// We've already sent the join packet, we have nothing to do.
+    Joined,
+    /// We want to leave the group, we have to send a leave packet.
+    Leaving,
+}
+
```

Comment:
Shouldn't we move this to `interface/igmp.rs`?

Reply:
it doesn't work because `multicast_groups` exists if `any(feature = "proto-igmp", feature = "proto-ipv6")`, but `mod igmp` exists if `feature = "proto-igmp"` only. So it would break if you enable ipv6 but not igmp.

(fwiw I think these feature flags are a bit confusing. I'd like to refactor this so there's a single `multicast` Cargo feature that enables multicast for ipv4 and/or ipv6 depending on what you have enabled. Then rename `mod igmp` to `mod multicast` and move all multicast-related stuff there. I'd rather do this in a next PR tho)

Reply:
I was thinking about the same thing.

Reply:
got nerdsniped, just did it in #988